### PR TITLE
style: add godoc to the armtype structs used in the template refactor

### PR DESCRIPTION
**Pull request by tariq1890**
Adding godoc to the publicly exposed types :).

### CODE REVIEW

File: pkg/engine/armtype.go [hunk 1]
```diff
@@ -29,86 +30,103 @@ func (arm ARMResource) MarshalJSON() ([]byte, error) {
...
 	sysauth.RoleAssignment
 }
 
+// VirtualNetworkARM embeds the ARMResource type in network.VirtualNetwork.
 type VirtualNetworkARM struct {
 	ARMResource
 	network.VirtualNetwork
 }
 
+// NetworkSecurityGroupARM embeds the ARMResource type in network.NetworkSecurityGroupARM.
```
Comment (sozercan) [inline pkg/engine/armtype.go]:
```suggestion
// NetworkSecurityGroupARM embeds the ARMResource type in network.SecurityGroup.
```

File: pkg/engine/armtype.go [hunk 2]
```diff
@@ -29,86 +30,103 @@ func (arm ARMResource) MarshalJSON() ([]byte, error) {
...
 	network.LoadBalancer
 }
 
+// NetworkInterfaceARM embeds the ARMResource type in network.Interface.
 type NetworkInterfaceARM struct {
 	ARMResource
 	network.Interface
 }
 
+// DocumentDBAccountARM embeds the ARMResource type in 	documentdb.DatabaseAccountCreateUpdateParameters.
```
Comment (sozercan) [inline pkg/engine/armtype.go]:
```suggestion
// DocumentDBAccountARM embeds the ARMResource type in documentdb.DatabaseAccountCreateUpdateParameters.
```

File: pkg/engine/armtype.go (hunk 1 above)
Comment (tariq1890) [reply pkg/engine/armtype.go]: Wooh! Thanks :)

File: pkg/engine/armtype.go (hunk 2 above)
Comment (tariq1890) [reply pkg/engine/armtype.go]: Whoops.

**Comment by tariq1890** [comment]
@sozercan Thanks! Review comments addressed.

**Pull request merged**